Filtered_kernel/include/CGAL/Lazy.h

```diff
@@ -35,8 +35,11 @@
 #include <CGAL/Bbox_2.h>
 #include <CGAL/Bbox_3.h>
 #include <vector>
+#include <type_traits>
 #include <CGAL/Default.h>
 #include<CGAL/tss.h>
+#include <CGAL/is_iterator.h>
+#include <CGAL/transforming_iterator.h>
```

Comment:
This include of `<CGAL/transforming_iterator.h>` introduced a dependency to `NewKernel_d`. Could this header be moved to `STL_Extension`, and even merged in `<CGAL/iterator.h>`?

Reply:
Oh yes, I should move it. Not sure about merging it though.

Reply:
Having a look at `CGAL::transforming_iterator`, I have two remarks...

  - In `<CGAL/transforming_iterator.h>`, `<CGAL/transforming_pair_iterator.h>`, and `<CGAL/typeset.h>`, you do not follow the CGAL naming guidelines, of having types named starting with a capital letter. I imagine you embraced the coding convention of the STL. I kind of agree with that convention for CGAL things that belong to STL_Extension. Would you like to propose a change of our coding convention?

  - I have seen usages of `transforming_iterator` is a few headers without the corresponding `#include`. You should run `clang/gcc -fsyntax-only` on your headers. The `CHECK` part of our Travis checks only verify the documented headers... and in `NewKernel_d` that means only `<CGAL/Epick_d.h>` and `<CGAL/Epeck_d.h>`.

Reply:
I don't mind adding a capital letter if that helps...
I guess I could add more #includes, but only the 2 documented headers really matter.

Reply:
> Not sure about merging it though.

Any good reason for not merging it?

Reply:
Any good reason why packages are split in several files instead of having everything in one huge file?

Reply:
Actually, I do not really care about the naming with capital letter or not. As long as it belongs to a kind of extension to the STL, I would agree that we can following the STL naming conventions.

What really matters is the move of the header <CGAL/transforming_iterator.h>, or its code, to STL_Extension, to avoid adding a dependency.